apps/vaults/hooks/useSolverCowswap.tsx

```diff
@@ -113,20 +115,53 @@ export function useSolverCowswap(): TSolverContext {
 	** call getQuote to get the current quote for the provided request.current.
 	**********************************************************************************************/
 	const init = useCallback(async (_request: TInitSolverArgs, shouldLogError?: boolean): Promise<TNormalizedBN> => {
-		if (isDisabled || !_request.inputToken.solveVia?.includes(Solver.COWSWAP)) {
+		/******************************************************************************************
+		** First we need to know which token we are selling to the zap. When we are depositing, we
+		** are selling the inputToken, when we are withdrawing, we are selling the outputToken.
+		** based on that token, different checks are required to determine if the solver can be
+		** used.
+		******************************************************************************************/
+		const sellToken = _request.isDepositing ? _request.inputToken: _request.outputToken;
```

Comment:
nit: space for readability
```suggestion
		const sellToken = _request.isDepositing ? _request.inputToken : _request.outputToken;
```